pkg/piecestore/psserver/bucket_refresher.go

```diff
@@ -0,0 +1,80 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package psserver
+
+import (
+	"context"
+	"flag"
+	"time"
+
+	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+
+	"storj.io/storj/pkg/dht"
+	"storj.io/storj/pkg/pb"
+)
+
+// Error is a standard error class for this package.
+var (
+	Error                = errs.Class("kademlia bucket refresher error")
+	defaultCheckInterval = flag.Duration("piecestore.kbucket-refresher.check-interval", time.Hour, "number of seconds to sleep between updating the kademlia bucket")
+)
+
+// refreshService contains the information needed to run the bucket refresher service
+type refreshService struct {
+	logger *zap.Logger
+	rt     dht.RoutingTable
+	server *Server
+}
+
+func newService(logger *zap.Logger, rt dht.RoutingTable, server *Server) *refreshService {
+	return &refreshService{
+		logger: logger,
+		rt:     rt,
+		server: server,
+	}
+}
+
+// Run runs the bucket refresher service
+func (service *refreshService) Run(ctx context.Context) (err error) {
```

Comment:
Is this logic only storagenode specific or does satellite need to implement this as well, or does it implement it differently?